File: biblioteca-web/src/app/models/reservation.model.ts
export interface Reservation {
  id: number;
  folio: string;

  bookId: number;
  bookTitle: string;
  author: string;

  studentName: string;
  matricula: string;
  userRole: string;

  requestDate: string;
  requestTime: string;
  expiresAt: string;

  status: 'pendiente' | 'entregado' | 'expirada' | 'cancelada';
}

File: backend/biblioteca-api/app/models/reservation_model.py
from typing import Literal
from pydantic import BaseModel


class ReservationCreate(BaseModel):
    id: int
    folio: str

    bookId: int
    bookTitle: str
    author: str

    studentName: str
    matricula: str
    userRole: str

    requestDate: str
    requestTime: str
    expiresAt: str

    status: Literal["pendiente", "entregado", "expirada", "cancelada"] = "pendiente"